File: 25. Check balanced tree.cpp
#include <bits/stdc++.h>
using namespace std;
struct Node
{
    int data;
    Node* left;
    Node* right;
    Node(int d) : data(d), left(nullptr), right(nullptr) {}
};
int checkHeight(Node* root)
{
    if (!root)
        return 0;
    int lh = checkHeight(root->left);
    if (lh == -1)
        return -1;
    int rh = checkHeight(root->right);
    if (rh == -1)
        return -1;
    if (abs(lh - rh) > 1)
        return -1;
    return 1 + max(lh, rh);
}
int main()
{
    Node* root = new Node(1);
    root->left = new Node(2);
    root->left->left = new Node(3);
    cout << (checkHeight(root) != -1 ? "Balanced" : "Not Balanced") << endl;
    return 0;
}

File: 25. Check balanced tree.py
class Node:
    def __init__(self,data): self.data=data; self.left=None; self.right=None

def is_balanced(root):
    def helper(node):
        if not node: return 0
        lh=helper(node.left)
        if lh==-1: return -1
        rh=helper(node.right)
        if rh==-1: return -1
        if abs(lh-rh)>1: return -1
        return 1+max(lh,rh)
    return helper(root)!=-1

if __name__=='__main__':
    root=Node(1); root.left=Node(2); root.left.left=Node(3)
    print('Balanced' if is_balanced(root) else 'Not Balanced')
